src/test/java/io/airlift/compress/AbstractTestCompression.java

```diff
@@ -142,6 +148,89 @@ public void testDecompressionBufferOverrun(DataSet dataSet)
         assertByteArraysEqual(padding, 0, padding.length, uncompressed, uncompressed.length - padding.length, padding.length);
     }
 
+    @Test
+    public void testDecompressInputBoundsChecks()
+    {
+        byte[] data = new byte[1024];
+        new Random(1234).nextBytes(data);
+        Compressor compressor = getCompressor();
+        byte[] compressed = new byte[compressor.maxCompressedLength(data.length)];
+        int compressedLength = compressor.compress(data, 0, data.length, compressed, 0, compressed.length);
+
+        Decompressor decompressor = getDecompressor();
+        Throwable throwable;
+
+        // null input buffer
+        assertThatThrownBy(() -> decompressor.decompress(null, 0, compressedLength, data, 0, data.length))
+                .isInstanceOf(NullPointerException.class);
+
+        // mis-declared buffer size
+        byte[] compressedChoppedOff = Arrays.copyOf(compressed, compressedLength - 1);
+        throwable = catchThrowable(() -> decompressor.decompress(compressedChoppedOff, 0, compressedLength, data, 0, data.length));
+        if (throwable instanceof UncheckedIOException) {
```

Comment:
Who throws this exception? In general, there shouldn’t be any IO related exceptions involved.

Reply:
GZIP code under the hood, AFAIR

(the expected exceptions are "whatever is looks somewhat reasonable", the fact that there is _an_ exception is the most important part)